reproducible/Dockerfile.nix

```diff
@@ -0,0 +1,185 @@
+                chmod -R a+rw \$src \n\
+                mkdir -p \$out \n\
+                cd \$out \n\
+                \$OE_SIM cmake -G \"Unix Makefiles\" \$src -DCMAKE_BUILD_TYPE=RelWithDebInfo  \n\
+            ''; \n\
+  \n\
+        buildPhase = '' \n\
+                echo \$OE_SIMULATION \n\
+                make VERBOSE=1 -j 4 \n\
+            ''; \n\
+        checkPhase = '' \n\
+                /* We must include something in the check phase or it will default */ \n\
+                echo \"ctest performed in nix-shell \" \n\
+            ''; \n\
+\n\
+        installPhase = '' \n\
+                echo \"install phase skipped \" \n\
+            ''; \n\
+\n\
+        fixupPhase = '' \n\
+                echo \"fixup phase skipped \" \n\
+            ''; \n\
+        \n\
+        shellHook = '' \n\
+\n\
+                cd \$out\n\
+                chmod -Rf a+w .\n\
+\n\
+                echo \"=== ctest -E samples\|oegdb-test\|report\" \n\
+                if \$DO_CHECK\n\
+                then \n\
+                    find ./tests -type d -exec chmod a+w {} \;\n\
+\n\
+                    LD_LIBRARY_PATH=/home/azureuser/.nix_libs \$OE_SIM ctest -E samples\|oegdb-test\|report \n\
+                    CTEST_RESULT=\$?\n\
+                    if [ \$CTEST_RESULT -ne 0]\n\
+                    then\n\
+                        echo \"ERROR: Ctests failed\"\n\
+                        exit \$CTEST_RESULT\n\
+                    fi\n\
+                fi\n\
+\n\
+                if [ \$(uname -m) == \"aarch64\" ]\n\
+                then \n\
+                    LD_INTERPRETER=\"/lib/aarch64-linux-gnu/ld-linux-aarch64.so.1\"\n\
+                elif [ \$(uname -m) == \"x86_64\" ]\n\
+                then\n\
+                    LD_INTERPRETER=\"/lib64/ld-linux-x86-64.so.2\"\n\
+                else\n\
+                    LD_INTERPRETER=\"UNSUPPORTED ARCHITECTURE\"\n\
+                fi\n\
+                echo \"=== FIXUP \$LD_INTERPRETER\"\n\
+                find \$out -type f -executable -exec patchelf --set-interpreter \$LD_INTERPRETER {} \;\n\
+                find \$out -type f -executable -exec patchelf --remove-rpath {} \;\n\
+\n\
+		if [ "$DEB_PACKAGE" != 'false' ] \n\
+                then\n\
+                    /home/azureuser/build_deb_pkg.sh\n\
+                else \n\
+                    echo "skipping package build" \n\
+                fi\n\
+                if \$INTERACTIVE_SHELL\n\
+                then\n\
+                    echo \"=== Complete\"\n\
+                    exit 0\n\
+                fi \n\
+            '';  \n\
+}  \n\
+" > /home/$BUILD_USER/shell.nix
+
+
+
+RUN echo "User is $USER "
+#install the required software
+#RUN touch .bash_profile \
+#
+# We add the nix install and packages into the container rather than waiting for run time.
+# The packages are then located in the nix store until the next push of the container
+#RUN curl https://nixos.org/releases/nix/nix-2.3.7/install | /bin/bash
```

Comment:
Is nix-2.3.7 installed or this line is really commented out?

Reply:
2.3.7 is installed, but rather than blindly getting a script from the net, we include it in the git repo.

Reply:
OK, I got it. This line is commented out and should be removed. The installation is handled in "install-nix.sh"

Reply:
line removed